### ISSUE TITLE
[help wanted] Securing an endpoint using Wildfly-Swarm, Keycloak and MP-JWT-Auth

### ISSUE THREAD

**Issue by DennisBayer** [eclipse/microprofile-jwt-auth#89]
Hi there,

I hope this is the right place to raise my question. Currently I am having some problems to get my example working. The authentication works fine, but the authorization fails, because the roles are omitted during the request.

Thus it's mostly easier to have a reproducible example than writing a wall of text, I created a [project](https://github.com/DennisBayer/playground-mp-jwt-auth/).

I'd appreciate any hints pointing out what is going wrong.

Thanks in advance and best regards
Dennis

**Comment by OndrejM** [comment]
Hi,

According to the logs you seem to get roles passed from the JWT token but not propagated through keycloak. This is an implementation detail and in regards to the MicroProfile specification you are probably doing all things correctly.

Better ask KeyCloak or Wildfly Swarm maintainers what may be wrong in your setup.

**Comment by starksm64** [comment]
I know that Keycloak was not following the MP-JWT format for the token, and I had created an issue to provide support for that.

**Comment by DennisBayer** [comment]
Hi @starksm64 ,

are you referring to the missing upn and group node? I think I stumbled over your issue. Someone mentioned that mappers could to the job atm. Therefore I added two mappers within keycloak to add both. See https://github.com/DennisBayer/playground-mp-jwt-auth/#example. So the format of my used token should be ok?! 

Or do you assume that `org.keycloak.adapters.wildfly.WildflyRequestAuthenticator` or `org.keycloak.adapters.RequestAuthenticator` removes the entries "upn" and "group" (https://github.com/DennisBayer/playground-mp-jwt-auth/#output-for-playgroundsecured) and your ticket addressed this part of the flow?

(btw: If someone is interested what informations the keycloak user mailing list might reveal ;) : http://lists.jboss.org/pipermail/keycloak-user/2018-April/013790.html)

**Comment by DennisBayer** [comment]
Hi there,

I did some debugging and might have found the issue. It seems that the keycloak logic has some issues with passing the roles (I added an update to the project readme).

If this is the cause this issue could be closed.

Does anybody here know some contacts in the keycloak project to poke someone since https://issues.jboss.org/browse/KEYCLOAK-7309 did not any attention so far? ;)

Regards
